pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -26,142 +29,172 @@ func (g OutboundProxyGenerator) Generate(ctx xds_context.Context, proxy *model.P
-		}
-		if destination.Weight == 0 {
-			// Envoy doesn't support 0 weight
-			continue
+func (_ OutboundProxyGenerator) generateLDS(proxy *model.Proxy, subsets []envoy_common.ClusterSubset, outbound *kuma_mesh.Dataplane_Networking_Outbound, protocol mesh_core.Protocol) (*envoy_api_v2.Listener, error) {
+	oface, err := proxy.Dataplane.Spec.Networking.ToOutboundInterface(outbound)
+	if err != nil {
+		return nil, err
+	}
+
+	meshName := proxy.Dataplane.Meta.GetMesh()
+	sourceService := proxy.Dataplane.Spec.GetIdentifyingService()
+	serviceName := outbound.GetTagsIncludingLegacy()[kuma_mesh.ServiceTag]
+	outboundListenerName := envoy_names.GetOutboundListenerName(oface.DataplaneIP, oface.DataplanePort)
+	filterChainBuilder := func() *envoy_listeners.FilterChainBuilder {
+		filterChainBuilder := envoy_listeners.NewFilterChainBuilder()
+		switch protocol {
+		case mesh_core.ProtocolHTTP:
+			// configuration for HTTP case
+			filterChainBuilder.
+				Configure(envoy_listeners.HttpConnectionManager(serviceName)).
+				Configure(envoy_listeners.Tracing(proxy.TracingBackend)).
+				Configure(envoy_listeners.HttpAccessLog(meshName, envoy_listeners.TrafficDirectionOutbound, sourceService, serviceName, proxy.Logs[serviceName], proxy)).
+				Configure(envoy_listeners.HttpOutboundRoute(envoy_names.GetOutboundRouteName(serviceName)))
+		case mesh_core.ProtocolTCP:
+			fallthrough
+		default:
+			// configuration for non-HTTP cases
+			filterChainBuilder.
+				Configure(envoy_listeners.TcpProxy(serviceName, subsets...)).
+				Configure(envoy_listeners.NetworkAccessLog(meshName, envoy_listeners.TrafficDirectionOutbound, sourceService, serviceName, proxy.Logs[serviceName], proxy))
 		}
-		clusters = append(clusters, envoy_common.ClusterInfo{
-			Name:   envoy_names.GetDestinationClusterName(service, destination.Destination),
-			Weight: destination.Weight,
-			Tags:   destination.Destination,
-		})
+		return filterChainBuilder
+	}()
+	listener, err := envoy_listeners.NewListenerBuilder().
+		Configure(envoy_listeners.OutboundListener(outboundListenerName, oface.DataplaneIP, oface.DataplanePort)).
+		Configure(envoy_listeners.FilterChain(filterChainBuilder)).
+		Configure(envoy_listeners.TransparentProxying(proxy.Dataplane.Spec.Networking.GetTransparentProxying())).
+		Build()
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not generate listener %s for service %s", outboundListenerName, serviceName)
 	}
-	return
+	return listener, nil
 }
 
-func (_ OutboundProxyGenerator) generateEds(ctx xds_context.Context, proxy *model.Proxy, clusters []envoy_common.ClusterInfo) (resources []*model.Resource, allEndpoints []model.Endpoint, _ error) {
-	for _, cluster := range clusters {
-		serviceName := cluster.Tags[kuma_mesh.ServiceTag]
+func (_ OutboundProxyGenerator) generateCDS(ctx xds_context.Context, proxy *model.Proxy, clusters envoy_common.Clusters) (model.ResourceSet, error) {
```

Comment:
When it comes to acronyms I thought we always make just the first letter capital. But I know that according to a convention - https://github.com/golang/go/wiki/CodeReviewComments#initialisms, letters should be in a consistent case. So do we want to follow it everywhere now?

Reply:
Yes, that's why I changed `generateEds` to `generateEDS`